subprojects/core/src/integTest/groovy/org/gradle/api/tasks/DeferredTaskDefinitionIntegrationTest.groovy

```diff
@@ -1089,4 +1082,32 @@ class DeferredTaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         and:
         executed ":foo", ":baz", ":fizz", ":fuzz", ":some"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/6319")
+    def "can use getTasksByName from a lazy configuration action"() {
+        settingsFile << """
+            include "sub"
+        """
+        buildFile << """
+            plugins {
+                id 'base'
+            }
+            tasks.whenTaskAdded {
```

Comment:
Help me understand - why is this important for this test?  Why not simply execute `foo` and validate that when it's realized we don't get an exception?

Reply:
It changes the order of project evaluation so that the root project and subprojects are evaluated first and then tasks are realized and then this problem doesn't occur because the subproject was already evaluated by the time we get to `getTasksByName`.  This forces subprojects to be evaluated as part of evaluating the root project while in the register call.

Reply:
👍